cpg_infra/driver.py

```diff
@@ -960,7 +960,40 @@ def setup_container_registry(self):
         + cpg-common container registries
         :return:
         """
+        self.setup_dataset_container_registry()
+        self.setup_inherited_container_registries()
 
+    def setup_dataset_container_registry(self):
+        """
+        If required, setup a container registry for a dataset
+        :return:
+        """
+        if not self.dataset_config.create_container_registry:
+            return
+
+        # mostly because this current format requires the project_id
+        custom_container_registry = self.infra.create_container_registry('images')
+        for kind, account in self.access_level_groups:
+            self.infra.add_member_to_container_registry(
+                f'test-images-reader-in-container-registry',
+                registry=custom_container_registry,
+                member=account,
+                membership=ContainerRegistryMembership.READER,
+            )
+            if kind in ('standard', 'full'):
+                self.infra.add_member_to_container_registry(
+                    f'{kind}-images-writer-in-container-registry',
+                    registry=custom_container_registry,
+                    member=account,
+                    membership=ContainerRegistryMembership.APPEND,
+                )
+
+    def setup_inherited_container_registries(self):
```

Comment:
If this is mostly meant to handle our existing setup temporarily, maybe "legacy" is the better description?

```suggestion
    def setup_legacy_container_registries(self):
```

If you want to change the implementation later to support access to repositories of dataset dependencies, maybe `dependent` could be a better description?

Reply:
Yep agreed, renamed to legacy.